readthedocs/projects/admin.py

```diff
@@ -177,6 +186,61 @@ def delete_selected_and_artifacts(self, request, queryset):
                 broadcast(type='app', task=remove_dir, args=[project.doc_path])
         return delete_selected(self, request, queryset)
 
+    def mark_as_abandoned(self, request, queryset):
+        """
+        Marks selected projects as abandoned.
+
+        It adds '-abandoned' to the slug, marks
+        Project.is_abandoned and sends an email to the user
+        notifying the change and link to the new project/docs
+        page.
+        """
+        # Change project's slug and mark it as abandoned.
+        queryset.update(
+            slug=Concat('slug', Value('-abandoned')),
+            is_abandoned=True
+        )
+
+        # Notifying users via email.
+        qs_iterator = queryset.iterator()
+        for project in qs_iterator:
+            users = project.users.get_queryset()
+            for user in users:
+                new_proj_url = '{}{}'.format(settings.PRODUCTION_DOMAIN, project.get_absolute_url())
+                send_email(
```

Comment:
I think this is the place where @agjohnson says that can be replaced by our current notification system: it sends emails to users and also add a notification _for those users_. As you are doing here with `send_email` and `self.message_user`

Reply:
Actually `self.message_user` doesn't send a notification to that user, it is for django-admin panel messages, only for the admin user. These ones -
![screenshot from 2018-12-20 21-43-04](https://user-images.githubusercontent.com/29149191/50296478-930ea600-04a0-11e9-9c1e-15a9df0ce70f.png)

And for the send_mail, I will try to integrate the Notification system.